app/src/test/java/org/mozilla/fenix/tabstray/TabsTrayFragmentTest.kt

```diff
@@ -337,13 +322,19 @@ class TabsTrayFragmentTest {
         val tabLayout: TabLayout = mockk {
             every { getTabAt(any()) } returns tab
         }
-        every { fragment.tab_layout } returns tabLayout
-        every { fragment.tabsTray } returns tabsTray
 
-        fragment.selectTabPosition(2, true)
+        mockkStatic(ViewBindings::class) {
+            every { ViewBindings.findChildViewById<View>(tabsTrayBinding.root, tabsTrayBinding.tabsTray.id) } returns tabsTray
+            every { ViewBindings.findChildViewById<View>(tabsTrayBinding.root, tabsTrayBinding.tabLayout.id) } returns tabLayout
```

Comment:
We saw issues mocking the response of ViewBinding properties or verifying when they're accessed but were able to workaround those cases.
Here we needed to test exactly this scenarios so after many tries this seems to be the simplest and cleanest approach for replacing ViewBinding properties with mocks.

Other approach that works is creating a new ViewBinding instance by calling the private constructor through reflection and passing the mocks as parameters but this will need all parameters to be provided and and the number can quickly grow.

Also looked into using `bind` to have the ViewBinding constructed for us and we could leverage the inflated layout in which we replace the Views we want mocked but the assignments done in this method were throwing cast exceptions as `ViewBindings.findChildViewById` always returned a simple `View`,